core/src/main/scala/org/bykn/bosatsu/DefRecursionCheck.scala

```diff
@@ -398,9 +398,9 @@ object DefRecursionCheck {
      */
     def checkDef[A](state: State, defstmt: DefStatement[Pattern.Parsed, (OptIndent[Declaration], A)]): Res = {
       val body = defstmt.result._1.get
-      val args = defstmt.args.flatMap(_.names)
-      val state1 = state.inDef(defstmt.name, args)
-      checkForIllegalBinds(state, defstmt.name :: args, body) {
+      val nameArgs = defstmt.args.flatMap(_.names)
```

Comment:
this was just totally wrong. This creates a list of names, but it isn't the parameters in order.